src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.Small.cs

```diff
@@ -92,31 +90,23 @@ protected override void InitializeConstructorArgumentCaches(ref ReadStack state,
             for (int i = 0; i < typeInfo.ParameterCount; i++)
             {
                 JsonParameterInfo parameterInfo = cache[i].Value;
-
-                // We can afford not to set default values for ctor arguments when we should't deserialize because the
-                // type parameters of the `Arguments` type provide default semantics that work well with value types.
-                if (parameterInfo.ShouldDeserialize)
```

Comment:
This check is incorrect -- we need to populate the default constructor values even for ignored properties. This is changed to be consistent with the large constructor converter.

Reply:
I'd rather do opposite and not do it for large constructors but fine with me to leave this as is in this PR since there is already issue open

Reply:
(also seems that small is more common case so I'd not want to reinforce that)

Reply:
Fundamentally this is what causing the bug in https://github.com/dotnet/runtime/issues/60082 so I think we should fix it.